lib/addons/helm-addon/index.ts

```diff
@@ -1,26 +1,66 @@
 import { Duration } from "aws-cdk-lib";
 import { Construct } from "constructs";
 import * as spi from "../..";
-import { cloneDeep } from "../../utils";
+import { cloneDeep, ConstraintsType, StringConstraint, UrlStringConstraint, validateConstraints } from "../../utils";
 import { HelmChartConfiguration, KubectlProvider } from "./kubectl-provider";
 
 export type HelmAddOnProps = HelmChartConfiguration;
 export type HelmAddOnUserProps = Partial<HelmChartConfiguration>;
 
+export const helmAddonPropsContraints: ConstraintsType<HelmAddOnProps> = {
+   /**
+   * chart can be no less than 1 character long, and no greater than 63 characters long.
+   * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+   */
+    chart: new StringConstraint(1, 63),
+
+   /**
+   * name can be no less than 1 character long, and no greater than 63 characters long.
+   * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+   */
+    name: new StringConstraint(1, 63),
+
+   /**
+   * namespace can be no less than 1 character long, and no greater than 63 characters long.
+   * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+   */
+    namespace: new StringConstraint(1, 63),
+
+   /**
+   * release can be no less than 1 character long, and no greater than 63 characters long.
+   * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+   */
+    release: new StringConstraint(1, 63),
+
+   /**
+   * repository can be no less than 1 character long, and no greater than 63 characters long. It also must follow a URL format.
+   * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+   */
+    repository: new UrlStringConstraint(1, 63),
+    
+   /**
+   * version can be no less than 1 character long, and no greater than 63 characters long.
+   * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+   */
+    version: new StringConstraint(1, 63)
+};
+
 export abstract class HelmAddOn implements spi.ClusterAddOn {
 
     props: HelmAddOnProps;
 
     constructor(props: HelmAddOnProps) {
         this.props = cloneDeep(props); // avoids polution when reusing the same props across stacks, such as values
+
+        validateConstraints([props], helmAddonPropsContraints, HelmAddOn.name);
```

Comment:
why are you passing props as [props]? Your code is expected to handle arrays or individual objects, correct?

Reply:
Oops thanks for catching that, that was from testing but forgot to change it in helm like I did the others.